src/dtgtk/range.c

```diff
@@ -1738,6 +1744,37 @@ gchar *dtgtk_range_select_get_bounds_pretty(GtkDarktableRangeSelect *range)
   return txt;
 }
 
+gchar *dtgtk_range_select_get_rating_bounds_pretty(GtkDarktableRangeSelect *range)
+{
+  if((range->bounds & DT_RANGE_BOUND_MIN)) 
+    range->select_min_r = range->min_r;
+  if((range->bounds & DT_RANGE_BOUND_MAX)) 
+    range->select_max_r = range->max_r;
+    
+  if(range->select_min_r == range->select_max_r)
+    return g_strdup_printf("%s %s", range->print(range->select_min_r, TRUE), _("only"));
+
+  int rating_min = (int)floor(range->select_min_r);
+  int rating_max = (int)floor(range->select_max_r);
+
+  if(rating_min == -1 && rating_max == 0)
+    return g_strdup_printf("%s + %s", _("rejected"), _("not rated"));
+
+  if(range->bounds & DT_RANGE_BOUND_MIN)
+    return g_strdup_printf("≤%s + %s", range->print(range->select_max_r, TRUE), _("rejected"));
+  else if(range->bounds & DT_RANGE_BOUND_MAX)
+  {
+    if(rating_min == 0)
+      return g_strdup(_("all except rejected"));
+    else
+      return g_strdup_printf("≥%s", range->print(range->select_min_r, TRUE));    
+  }
+  else if(rating_min == 0)
+    return g_strdup_printf("≤%s", range->print(range->select_max_r, TRUE));
+
+  return dtgtk_range_select_get_bounds_pretty(range);
```

Comment:
is the returned value a duplicate string? I mean you stddup() all strings above, so free them after, so dtgtk_range_select_get_bounds_pretty() must return a strdup() string.

Reply:
sorry, don't understand that one...
what am I doing wrong?